cmake/kokkos_functions.cmake

```diff
@@ -60,6 +60,428 @@ FUNCTION(kokkos_option CAMEL_SUFFIX DEFAULT TYPE DOCSTRING)
+    KOKKOS_APPEND_CONFIG_LINE("ENDIF()")
+  ENDIF()
+  KOKKOS_APPEND_CONFIG_LINE("FIND_DEPENDENCY(${NAME})")
+ENDMACRO()
+
+MACRO(kokkos_export_imported_tpl NAME)
+  IF (NOT KOKKOS_HAS_TRILINOS)
+    GET_TARGET_PROPERTY(LIB_TYPE ${NAME} TYPE)
+    IF (${LIB_TYPE} STREQUAL "INTERFACE_LIBRARY")
+      # This is not an imported target
+      # This an interface library that we created
+      INSTALL(
+        TARGETS ${NAME}
+        EXPORT KokkosTargets
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      )
+    ELSE()
+      #make sure this also gets "exported" in the config file
+      KOKKOS_APPEND_CONFIG_LINE("ADD_LIBRARY(${NAME} UNKNOWN IMPORTED)")
+      KOKKOS_APPEND_CONFIG_LINE("SET_TARGET_PROPERTIES(${NAME} PROPERTIES")
+      
+      GET_TARGET_PROPERTY(TPL_LIBRARY ${NAME} IMPORTED_LOCATION)
+      IF(TPL_LIBRARY)
+        KOKKOS_APPEND_CONFIG_LINE("IMPORTED_LOCATION ${TPL_LIBRARY}")
+      ENDIF()
+
+      GET_TARGET_PROPERTY(TPL_INCLUDES ${NAME} INTERFACE_INCLUDE_DIRECTORIES)
+      IF(TPL_INCLUDES)
+        KOKKOS_APPEND_CONFIG_LINE("INTERFACE_INCLUDE_DIRECTORIES ${TPL_INCLUDES}")
+      ENDIF()
+
+      GET_TARGET_PROPERTY(TPL_COMPILE_OPTIONS ${NAME} INTERFACE_COMPILE_OPTIONS)
+      IF(TPL_COMPILE_OPTIONS)
+        KOKKOS_APPEND_CONFIG_LINE("INTERFACE_COMPILE_OPTIONS ${TPL_COMPILE_OPTIONS}")
+      ENDIF()
+
+      SET(TPL_LINK_OPTIONS)
+      IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.13.0")
+        GET_TARGET_PROPERTY(TPL_LINK_OPTIONS ${NAME} INTERFACE_LINK_OPTIONS)
+      ENDIF()
+      IF(TPL_LINK_OPTIONS)
+        KOKKOS_APPEND_CONFIG_LINE("INTERFACE_LINK_OPTIONS ${TPL_LINK_OPTIONS}")
+      ENDIF()
+
+      GET_TARGET_PROPERTY(TPL_LINK_LIBRARIES  ${NAME} INTERFACE_LINK_LIBRARIES)
+      IF(TPL_LINK_LIBRARIES)
+        KOKKOS_APPEND_CONFIG_LINE("INTERFACE_LINK_LIBRARIES ${TPL_LINK_LIBRARIES}")
+      ENDIF()
+      KOKKOS_APPEND_CONFIG_LINE(")")
+    ENDIF()
+  ENDIF()
+ENDMACRO()
+
+
+#
+# @MACRO: KOKKOS_IMPORT_TPL()
+#
+# Function that checks if a third-party library (TPL) has been enabled and calls `find_package`
+# to create an imported target encapsulating all the flags and libraries
+# needed to use the TPL
+#
+# Usage::
+#
+#   KOKKOS_IMPORT_TPL(
+#     <NAME>
+#     NO_EXPORT
+#     INTERFACE
+#
+#   ``NO_EXPORT``
+#
+#     If specified, this TPL will not be added to KokkosConfig.cmake as an export
+#
+#   ``INTERFACE``
+#
+#     If specified, this TPL will build an INTERFACE library rather than an
+#     IMPORTED target
+MACRO(kokkos_import_tpl NAME)
+  CMAKE_PARSE_ARGUMENTS(TPL
+   "NO_EXPORT;INTERFACE"
+   ""
+   ""
+   ${ARGN})
+  IF (TPL_INTERFACE)
+    SET(TPL_IMPORTED_NAME ${NAME})
+  ELSE()
+    SET(TPL_IMPORTED_NAME Kokkos::${NAME})
+  ENDIF()
+
+  IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12.0") 
+    CMAKE_POLICY(SET CMP0074 NEW)
+  ENDIF()
```

Comment:
Please comment why the policy needs to be set within the macro.

Reply:
Done.